src/verilog/verilog_preprocessor.cpp

```diff
@@ -603,9 +615,7 @@ void verilog_preprocessort::directive()
 
       if(it==defines.end())
       {
-        error().source_location = source_location;
-        error() << "unknown preprocessor directive \"" << text << "\"" << eom;
-        throw 0;
+        throw verilog_preprocessor_errort() << "unknown preprocessor directive \"" << text << "\"";
```

Comment:
I'd have a preference towards RAII: could the constructor please require (!) a string to be passed?

Reply:
The class meets the RAII criteria; in particular, all invariants are satisfied. It's like a container, and as such may be constructed empty.